HEP-event-capture/src/models/Form.ts

```diff
@@ -0,0 +1,64 @@
+import { ProgramStage, DataElement } from "../Dhis2Metadata";
+import { CustomForm } from "../components/CustomForm";
+import * as _ from "lodash";
+
+export interface FormData {
+    sections: FormSection[];
+}
+
+export interface FormSection {
+    title: string;
+    dataElements: DataElement[];
+    programStageId: string;
+}
+
+export const formCodes = {
+    formSection: "HEP_POLICY_SECTION",
+    sectionOrder: "SECTION_ORDER",
+    formDataElementGroup: "HEP_POLICY_EVENT_CAPTURE",
+    optionList: "USE_OPTION_FIELD",
+};
+
+export class Form {
+    public static getFormHtml(programStage: ProgramStage): string {
+        const formData = this.getFormData(programStage);
+        const formHtml = CustomForm({ formData });
+        return formHtml;
+    }
+
+    public static getFormData(programStage: ProgramStage): FormData {
+        const allDataElementGroups = programStage.programStageDataElements.map(
+            psde =>
+                psde.dataElement.dataElementGroups.filter(deg =>
+                    new RegExp("^" + formCodes.formSection).test(deg.code)
```

Comment:
Maybe `deg.code && deg.code.startsWith(formCodes.formSection)`. Also, it seems that `code` is not present in the response when not set. So it should be `code?: string` in the interface.